src/object.d

```diff
@@ -1641,6 +1643,11 @@ class Throwable : Object
         string toString() const;
     }
 
+    private uint _refcount;     // 0 : allocated by GC
```

Comment:
Just a thought.  Would there be a way to use a different vptr for reference-counted Throwable's?  If so, you could use virtual functions instead of adding a _refcount to every Throwable object.  Not sure if this design is better or worse (or if it's even feasible to implement) but it's something to consider I think.